arduino/cores/status.go

```diff
@@ -127,6 +127,33 @@ func (packages Packages) GetPlatformReleaseDiscoveryDependencies(release *Platfo
 	return res, nil
 }
 
+// GetPlatformReleaseMonitorDependencies returns the monitor releases needed by the specified PlatformRelease
+func (packages Packages) GetPlatformReleaseMonitorDependencies(release *PlatformRelease) ([]*ToolRelease, error) {
+	if release == nil {
+		return nil, fmt.Errorf(tr("release cannot be nil"))
+	}
+
+	res := []*ToolRelease{}
+	for _, monitor := range release.MonitorDependencies {
+		pkg, exists := packages[monitor.Packager]
+		if !exists {
+			return nil, fmt.Errorf(tr("package %s not found"), monitor.Packager)
+		}
+		tool, exists := pkg.Tools[monitor.Name]
+		if !exists {
+			return nil, fmt.Errorf(tr("tool %s not found"), monitor.Name)
+		}
+
+		// We always want to use the latest available release for monitors
+		latestRelease := tool.LatestRelease()
+		if latestRelease == nil {
+			return nil, fmt.Errorf(tr("can't find latest release of %s"), monitor.Name)
+		}
+		res = append(res, latestRelease)
+	}
+	return res, nil
+}
```

Comment:
This is identical to `GetPlatformReleaseDiscoveryDependencies` apart from line 137, we could make generic.

Reply:
Yes, also `MonitorDependencies` is identical to `DiscoveryDependencies`.

The problem is that `release.MonitorDependencies` is an array of `MonitorDependecy`... we can make it generic if we change the type of `release.DiscoveryDependencies` and `release.MonitorDependencies` to be the same type (for example reusing`ToolDependencies` for both) but this change starts a chain reaction... I've another branch ready for that but I prefer to push it in another PR for review and not delay this one.

Reply:
Oh right right, the type is different. Let's leave as it is now for the time being.